things-to-do

```diff
@@ -0,0 +1,19 @@
+array data type -> support for null (not empty!) array
+
+proper serialization of request depending on api_version (some versions serialize only some fields)
+
+request decoder: missing request types (right now we only have Produce, Fetch, etc.)
+
+requests in general - kafka 2.0 has 43 types of request (0.11 has ~30)
+	will it be okay to macro out some of the definitions?
+	compare https://github.com/apache/kafka/blob/2.0/clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitRequest.java#L271
```

Comment:
in general kafka/java code is terribly verbose 
https://github.com/apache/kafka/blob/2.0/clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitRequest.java#L271
trying to have something more concise, but still might need some legwork to handle all versions 
(see the `// Only for v1` in the link)